Test fixes

pdeffebach opened the pull request:
I was kind of annoyed at the DataFrames deprecations popping up while running tests. Changed to reflect new `getindex` and `setindex` rules.

andreasnoack commented:
Thanks for doing this. I think you'd then have to remove versions prior to 0.19 from the `compat` section of the Project.toml file.

pdeffebach commented:
Thanks. deleted prior versions and added 0.20.

nalimilan commented on test/runtests.jl:
```diff
@@ -23,7 +23,7 @@ linreg(x::AbstractVecOrMat, y::AbstractVector) = qr!(simplemm(x)) \ y
     lm1 = fit(LinearModel, @formula(OptDen ~ Carb), form)
     test_show(lm1)
-    @test isapprox(coef(lm1), linreg(form[:Carb], form[:OptDen]))
+    @test isapprox(coef(lm1), linreg(form[:,:Carb], form[:,:OptDen]))
```
```suggestion
    @test isapprox(coef(lm1), linreg(form.Carb, form.OptDen))
```

nalimilan commented on test/runtests.jl:
```diff
@@ -40,7 +40,7 @@ linreg(x::AbstractVecOrMat, y::AbstractVector) = qr!(simplemm(x)) \ y
     @test isapprox(aicc(lm1), -24.409684288095946)
     @test isapprox(bic(lm1), -37.03440588041178)
-    lm2 = fit(LinearModel, hcat(ones(6), 10form[:Carb]), form[:OptDen], true)
+    lm2 = fit(LinearModel, hcat(ones(6), 10form[:,:Carb]), form[:,:OptDen], true)
```
```suggestion
    lm2 = fit(LinearModel, hcat(ones(6), 10form.Carb), form.OptDen, true)
```

nalimilan commented on test/runtests.jl:
```diff
@@ -90,7 +90,7 @@ end
 ## Example from http://www.ats.ucla.edu/stat/r/dae/logit.htm
 admit = CSV.read(joinpath(glm_datadir, "admit.csv"))
-admit[:rank] = categorical(admit[:rank])
+admit[!,:rank] = categorical(admit[:,:rank])
```
```suggestion
admit.rank = categorical(admit.rank)
```

nalimilan commented on test/runtests.jl:
```diff
@@ -166,7 +166,7 @@ anorexia = CSV.read(joinpath(glm_datadir, "anorexia.csv"))
 @testset "Offset" begin
     gm6 = fit(GeneralizedLinearModel, @formula(Postwt ~ 1 + Prewt + Treat), anorexia,
-              Normal(), IdentityLink(), offset=Array{Float64}(anorexia[:Prewt]))
+              Normal(), IdentityLink(), offset=Array{Float64}(anorexia[:,:Prewt]))
```
```suggestion
              Normal(), IdentityLink(), offset=Array{Float64}(anorexia.Prewt))
```

nalimilan commented on test/runtests.jl:
```diff
@@ -184,7 +184,7 @@ end
 @testset "Normal LogLink offset" begin
     gm7 = fit(GeneralizedLinearModel, @formula(Postwt ~ 1 + Prewt + Treat), anorexia,
-              Normal(), LogLink(), offset=Array{Float64}(anorexia[:Prewt]), rtol=1e-8)
+              Normal(), LogLink(), offset=Array{Float64}(anorexia[:,:Prewt]), rtol=1e-8)
```
```suggestion
              Normal(), LogLink(), offset=Array{Float64}(anorexia.Prewt), rtol=1e-8)
```

nalimilan commented on test/runtests.jl:
```diff
@@ -272,7 +272,7 @@ admit_agr = DataFrame(count = [28., 97, 93, 55, 33, 54, 28, 12],
     for distr in (Binomial, Bernoulli)
         gm14 = fit(GeneralizedLinearModel, @formula(admit ~ 1 + rank), admit_agr, distr(),
-                   wts=Array(admit_agr[:count]))
+                   wts=Array(admit_agr[:,:count]))
```
```suggestion
                   wts=Array(admit_agr.count))
```

nalimilan commented on test/runtests.jl:
```diff
@@ -288,12 +288,12 @@ end
 admit_agr2 = DataFrame(Any[[61., 151, 121, 67], [33., 54, 28, 12], categorical(1:4)],
     [:count, :admit, :rank])
-admit_agr2[:p] = admit_agr2[:admit] ./ admit_agr2[:count]
+admit_agr2[!,:p] = admit_agr2[:,:admit] ./ admit_agr2[:,:count]
```
```suggestion
admit_agr2.p = admit_agr2.admit ./ admit_agr2.count
```

nalimilan commented on test/runtests.jl:
```diff
@@ -288,12 +288,12 @@ end
 @testset "Binomial LogitLink aggregated" begin
     gm15 = fit(GeneralizedLinearModel, @formula(p ~ rank), admit_agr2, Binomial(),
-               wts=admit_agr2[:count])
+               wts=admit_agr2[:,:count])
```
```suggestion
               wts=admit_agr2.count)
```

nalimilan commented on test/runtests.jl:
```diff
@@ -436,10 +436,10 @@ end
     # Prediction from DataFrames
     d = convert(DataFrame, X)
-    d[:y] = Y
+    d[!,:y] = Y
```
```suggestion
    d.y = Y
```

nalimilan commented on test/runtests.jl:
```diff
@@ -483,7 +482,7 @@ end
     @test GLM.issubmodel(othermod, bothmod)
 
-    d[:Sum] = d[:Treatment] + (d[:Other] .== 1)
+    d[!,:Sum] = d[:,:Treatment] + (d[:,:Other] .== 1)
```
```suggestion
    d.Sum = d.Treatment + (d.Other .== 1)
```

pdeffebach commented:
Added all the desired `getproperty` calls and tests passed. Also changed the getproperty calls for the named tuples in the tests because I never see `getindex` with named tuples anywhere else. This PR is ready to be merged.

nalimilan commented:
Can you drop Manifest? That may be the reason why CI failed on 1.0.

pdeffebach commented:
Dropped. Sorry about that. I seem to be accidentally creating a manifest.toml on a bunch of PRs and I'm not sure why. Will look into it.
